### ISSUE TITLE
Not possible to set the order of participants that are later created in Sequence Diagram

### ISSUE THREAD

**Issue by sidharthv96** [mermaid-js/mermaid#4707]
@Valentine14th @sidharthv96 I'd like to report an issue with this feature. It's not possible to set the order of participants that are later created.

Using the provided example, let's try to move Carl as the first participant, then Bob and Alice. This does not work with the following error:

```
Error: It is not possible to have actors with the same id, even if one is destroyed before the next is created. Use 'AS' aliases to simulate the behavior
```

```mermaid
sequenceDiagram
    %% the next line is breaking the diagram
    participant Carl
    participant Bob
    participant Alice
    Alice->>Bob: Hello Bob, how are you ?
    Bob->>Alice: Fine, thank you. And you?
    create participant Carl
    Alice->>Carl: Hi Carl!
    create actor D as Donald
    Carl->>D: Hi!
    destroy Carl
    Alice-xCarl: We are too many
    destroy Bob
    Bob->>Alice: I agree
```

Is there a way around this issue to have better control on the ordering of actors/participants?

_Originally posted by @habovh in https://github.com/mermaid-js/mermaid/issues/4466#issuecomment-1667848049_

**Comment by VishalMCF** [comment]
@sidharthv96 Hello Siddharth I am not able to understand the exact issue. What I did is that I copied and pasted the example and it gave me an error that you showed. But you have not mentioned the expected behavior. What I can think of related to expected behavior is that after writing this command -> "create participant Carl" Carl should be shown after Bob and Alice. Can you please confirm if my observation is correct or wrong? I have started navigating the codebase in detail and if I understand the bug exactly I would like to resolve the issue.

**Comment by VishalMCF** [comment]
sequenceDiagram
    %% the next line is breaking the diagram
    participant Carl
    participant Bob
    participant Alice
    create participant Ca as Carl

Here when I used the alias it showed me 4 participants. But as per your expectation without using an alias it should render only 3 participants and Carl should be positioned in the end. Am I right??

**Comment by sidharthv96** [comment]
@habovh is the actual reporter of the problem. I only moved it from comments to issue.

**Comment by habovh** [comment]
Thanks for the mention @sidharthv96, I missed the previous comments.

I agree my comment taken out of context from the original PR —and even then— might not be the best at explaining what's wrong.

Take the following, which works fine:
```mermaid
sequenceDiagram
    %% participant Carl
    participant Bob
    participant Alice
    Alice->>Bob: Hello Bob, how are you ?
    Bob->>Alice: Fine, thank you. And you?
    create participant Carl
    Alice->>Carl: Hi Carl!
    create actor D as Donald
    Carl->>D: Hi!
    destroy Carl
    Alice-xCarl: We are too many
    destroy Bob
    Bob->>Alice: I agree
```
#4466 works, and Carl is created in position 3 then destroyed through messages correctly.

Now, my goal was to use the `participant` instructions at the beginning to reorder participants but keep the order of the messages, to set Carl at position 1 and still have it created/destroyed with messages.

So to summarise, the expected behaviour is to end up with participants looking like this:

| Step | Participant 1 | Participant 2 | Participant 3 |
| --- | --- | --- | --- |
| 1 | | Create Bob | Create Alice |
| 2 | Create Carl | | |
| 3 | Destroy Carl | | |
| 4 | | Destroy Bob | |
| 5 | | | Destroy Alice |

**Comment by astroDimitrios** [comment]
I think this Issue of ordering participants also applies to the diagram I am trying to make:

```mermaid
sequenceDiagram
    autonumber
    participant GHM as origin main
    participant M as main
    GHM ->> M: #32;
    Note right of GHM: First time: git clone<br/>Then: git pull
    create participant F as feature
    M ->> F: git switch -c feature
    loop
        F ->> F: #32;
        Note over F: Develop Changes:<br/>git add<br/>git commit
    end
    create participant GHF as Origin Feature
    F -->> GHF: #32;
    Note left of F: git push
    GHF -->> GHM: Pull Request and then Merge
    GHM ->> M: #32;
    Note right of GHM: git pull
    box GitHub Remote
    participant GHM
    participant GHF
    end
    box local
    participant M
    participant F
    end
```

I would like to be able to have the origin feature actor (GHF) next to origin main (GHM) and in the same GitHub Remote box even though GHF is created as the last actor.

**Comment by dverbru** [comment]
@VishalMCF: in short, what we need is a way to define the order of all participants, including those which are created along the way. Currently, created participants always appear after all participants defined previously, which is particularly bad when most participants are declared at the top.
